src/libraries/System.Text.Json/tests/JsonNode/JsonObjectTests.cs

```diff
@@ -0,0 +1,512 @@
+            Assert.True(iCollection.Contains(kvp1));
+            Assert.True(iCollection.Contains(kvp2));
+            Assert.False(iCollection.Contains(new KeyValuePair<string, JsonNode?>("?", null)));
+
+            Assert.True(iCollection.Remove(kvp1));
+            Assert.Equal(1, jObject.Count);
+
+            Assert.False(iCollection.Remove(new KeyValuePair<string, JsonNode?>("?", null)));
+            Assert.Equal(1, jObject.Count);
+        }
+
+        [Fact]
+        public static void IsReadOnly()
+        {
+            ICollection<KeyValuePair<string, JsonNode?>> jObject = new JsonObject();
+            Assert.False(jObject.IsReadOnly);
+        }
+
+        [Fact]
+        public static void NullPropertyValues()
+        {
+            var jObject = new JsonObject();
+            jObject["One"] = null;
+            jObject.Add("Two", null);
+            Assert.Equal(2, jObject.Count);
+            Assert.Null(jObject["One"]);
+            Assert.Null(jObject["Two"]);
+        }
+
+        [Fact]
+        public static void NullPropertyNameFail()
+        {
+            var jObject = new JsonObject();
+            Assert.Throws<ArgumentNullException>(() => jObject.Add(null, JsonValue.Create(0)));
+            Assert.Throws<ArgumentNullException>(() => jObject[null] = JsonValue.Create(0));
+        }
+
+        [Fact]
+        public static void IEnumerable()
+        {
+            var jObject = new JsonObject();
+            jObject["One"] = 1;
+            jObject["Two"] = 2;
+
+            IEnumerable enumerable = jObject;
+            int count = 0;
+            foreach (KeyValuePair<string, JsonNode?> node in enumerable)
+            {
+                count++;
+            }
+
+            Assert.Equal(2, count);
+        }
+
+        [Fact]
+        public static void MissingProperty()
+        {
+            var options = new JsonSerializerOptions();
+            JsonObject jObject = JsonSerializer.Deserialize<JsonObject>("{}", options);
+            Assert.Null(jObject["NonExistingProperty"]);
+            Assert.False(jObject.Remove("NonExistingProperty"));
+        }
+
+        [Fact]
+        public static void IDictionary_KeyValuePair()
+        {
+            IDictionary<string, JsonNode?> jObject = new JsonObject();
+            jObject.Add(new KeyValuePair<string, JsonNode?>("MyProperty", 42));
+            Assert.Equal(42, jObject["MyProperty"].GetValue<int>());
+
+            Assert.Equal(1, jObject.Keys.Count);
+            Assert.Equal(1, jObject.Values.Count);
+        }
+
+        [Fact]
+        public static void Clear_ContainsKey()
+        {
+            var jObject = new JsonObject();
+            jObject.Add("One", 1);
+            jObject.Add("Two", 2);
+            Assert.Equal(2, jObject.Count);
+
+            Assert.True(jObject.ContainsKey("One"));
+            Assert.True(jObject.ContainsKey("Two"));
+            Assert.False(jObject.ContainsKey("?"));
+
+            jObject.Clear();
```

Comment:
would it make sense to repeat the assertions in lines 125-126 after the call to `Clear()` has been made?